File: 04_quicksort/c++/01_loop_sum.cpp
#include <iostream>

using namespace std;

int sum(int (&arr)[4]){
	int total = 0;
	for (int i = 0; i < (sizeof(arr)/sizeof(*arr)) ; ++i){
		total += arr[i];
	}
	return total;
}

int main(int argc, char *argv[]) {
	int arr[] = {1, 2, 3, 4};
	cout << sum(arr) << endl;	
}

File: 04_quicksort/java/01_loop_sum.java
import java.util.*;
import java.lang.*;
import java.io.*;


class grokkingAlgorithms
{	
	public static int sum(int[] arr){
		int total = 0;
		for (int i = 0; i < arr.length; ++i){
			total += arr[i];
		}
		return total;
	}

	public static void main (String[] args) throws java.lang.Exception
	{
		int arr[] = {1, 2, 3, 4};
		System.out.print(sum(arr)); 
	}
}